GoogleDataTransport/GDTCORLibrary/Public/GDTCORDatabase.h

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2019 Google
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <GoogleDataTransport/GDTCORSqlite.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+/** This class creates and operates sqlite3 databases. */
+@interface GDTCORDatabase : NSObject
+
+/** The user_version PRAGMA of the db. */
+@property(atomic) int userVersion;
+
+/** The serial queue on which the db operates. */
+@property(nonatomic, readonly) dispatch_queue_t dbQueue;
```

Comment:
nit: Exposing `dbQueue` creates an impression that the instance methods can be safe to run on it, but current implementation will deadlock if we call `runQuery: ...` from `dbQueue`. Would you like to remove it from the class interface of make a comment?

Reply:
Moved to private header.